### ISSUE TITLE
`27--01bbcpolice.powercoremedia.com` breaks dnsmasq

### ISSUE THREAD

**Issue by jtmcdole** [StevenBlack/hosts#2018]
When importing the latest list, and converting, `27--01bbcpolice.powercoremedia.com` turns into:

```
address=/27--01bbcpolice.powercoremedia.com/
```

Which looks fine at first, but dnsmasq fails to load with "error in <file> at line <line>".  I dug through the source code and it appears to fail DNS parsing [rfc 5891 section 4.2.3.1](https://datatracker.ietf.org/doc/html/rfc5891#section-4.2.3.1).

```
[4.2.3.1](https://datatracker.ietf.org/doc/html/rfc5891#section-4.2.3.1).  Hyphen Restrictions

   The Unicode string MUST NOT contain "--" (two consecutive hyphens) in
   the third and fourth character positions and MUST NOT start or end
   with a "-" (hyphen).
```

If I replace "--" with "-\\-" or "\\--", dnsmasq loads.

**Comment by StevenBlack** [comment]
Hi John @jtmcdole thanks for this.

we came across this in Issue #1930 — what do you think?

**Comment by jtmcdole** [comment]
Yeah, it's a library that dnsmasq is using to parse domain names. I didn't get into the library, but it follows the spec I listed. I'll update my scripts to add a \

**Comment by StevenBlack** [comment]
Thanks John @jtmcdole that's very interesting.

Seems that getting certs for those domains will be increasingly difficult. I wonder how that'll apply to some of the self-signing certificate providers.

In the end those domains will disappear, but they are evidently still out there in userland.

**Comment by jtmcdole** [comment]
For future humans that come here: if you're using sed to parse and transform configs to dnsmasq an, you can use the following pattern at the start to reject non-IDN2 compatible double-hyphen names:

```
SKIPBADNAMES='/^0.0.0.0 (.*\.)?[^x][^n]--/b'

sed -r -n "$SKIPBADNAMES; $YOUR; $OTHER; $MAGIC" > stevenblack.dnsmasq.conf
```

You should also (unlike past-me) use `cat your.config | dnsmaq --test -C -` to verify *before* shipping your configs... thus avoiding taking down your server :)
